sql/auto_install.sql

```diff
@@ -1,27 +1,101 @@
+-- *
+-- * Clean up the exisiting tables
+-- *
+-- *******************************************************/
+
 SET FOREIGN_KEY_CHECKS=0;
+
 DROP TABLE IF EXISTS `membershipextras_contribrecur_lineitem`;
 
 SET FOREIGN_KEY_CHECKS=1;
+-- /*******************************************************
+-- *
+-- * Create new tables
+-- *
+-- *******************************************************/
+
+-- /*******************************************************
+-- *
+-- * membershipextras_contribrecur_lineitem
+-- *
+-- * Implements a relationship between recurring contributions and line items, used to store the current values for pending installments.
+-- *
+-- *******************************************************/
 CREATE TABLE `membershipextras_contribrecur_lineitem` (
-  `id` int unsigned NOT NULL AUTO_INCREMENT COMMENT 'Discount Item ID',
-  `contribution_recur_id` int unsigned NOT NULL COMMENT 'ID of the recurring contribution.',
-  `line_item_id` int unsigned NOT NULL COMMENT 'ID of the line item related to the recurring contribution.',
-  `start_date` datetime NULL COMMENT 'Start date of the period for the membership/recurring contribution.',
-  `end_date` datetime NULL COMMENT 'End date of the period for the membership/recurring contribution.',
-  `auto_renew` datetime NULL COMMENT 'If the line-item should be auto-renewed or not.',
-
-  PRIMARY KEY (`id`),
-
-  UNIQUE INDEX `index_contribrecurid_lineitemid`(
-    contribution_recur_id,
-    line_item_id
-  ),
-
-  CONSTRAINT FK_membershipextras_contribrecur_lineitem_contribution_recur_id
-    FOREIGN KEY (`contribution_recur_id`) REFERENCES `civicrm_contribution_recur`(`id`)
-    ON DELETE CASCADE,
-
-  CONSTRAINT FK_membershipextras_contribrecur_lineitem_line_item_id
-    FOREIGN KEY (`line_item_id`) REFERENCES `civicrm_line_item`(`id`)
-    ON DELETE CASCADE
-);
+
+
+     `id` int unsigned NOT NULL AUTO_INCREMENT  COMMENT 'Discount Item ID',
+     `contribution_recur_id` int unsigned NOT NULL   COMMENT 'ID of the recurring contribution.',
+     `line_item_id` int unsigned NOT NULL   COMMENT 'ID of the line item related to the recurring contribution.',
+     `start_date` datetime NULL   COMMENT 'Start date of the period for the membership/recurring contribution.',
+     `end_date` datetime NULL   COMMENT 'End date of the period for the membership/recurring contribution.',
+     `auto_renew` tinyint NULL   COMMENT 'If the line-item should be auto-renewed or not.' 
+,
+        PRIMARY KEY (`id`)
+ 
+    ,     UNIQUE INDEX `index_contribrecurid_lineitemid`(
+        contribution_recur_id
+      , line_item_id
+  )
+  
+,          CONSTRAINT FK_membershipextras_contribrecur_lineitem_contribution_recur_id FOREIGN KEY (`contribution_recur_id`) REFERENCES `civicrm_contribution_recur`(`id`) ON DELETE CASCADE,          CONSTRAINT FK_membershipextras_contribrecur_lineitem_line_item_id FOREIGN KEY (`line_item_id`) REFERENCES `civicrm_line_item`(`id`) ON DELETE CASCADE  
+)    ;
+
+ 
```

Comment:
@16kilobyte, normally we don't alter autogenerated files like this one, but this is actually an exception. We normally edit these to clean them up a bit (indentation, styling and the like). The reason is sometimes there is more code in here, other than the one generated by civix, and we need to keep track of all those DB changes needed on either install or uninstall. Also, you can remove all the comments at the start, kind of leaving it as it was.

BTW, according to our coding standard, all files should end in a blank line, so please remember to add this once you do the fix.